hummingbot/strategy/pure_market_making/constant_size_sizing_delegate.pyx

```diff
@@ -36,20 +41,27 @@ cdef class ConstantSizeSizingDelegate(OrderSizingDelegate):
                                           object pricing_proposal):
         cdef:
             MarketBase market = market_info.market
+            object buy_fees
             double base_asset_balance = market.c_get_balance(market_info.base_currency)
             double quote_asset_balance = market.c_get_balance(market_info.quote_currency)
             double bid_order_size = self._order_size
             double ask_order_size = self._order_size
             bint has_active_bid = False
             bint has_active_ask = False
 
+        buy_fees = market.c_get_fee(market_info.base_currency, market_info.quote_currency,
+                                    OrderType.MARKET, TradeType.BUY,
+                                    bid_order_size, pricing_proposal.buy_order_prices[0])
+
         if market.name == "binance":
             bid_order_size = market.c_quantize_order_amount(market_info.symbol, self.order_size, pricing_proposal.buy_order_prices[0])
             ask_order_size = market.c_quantize_order_amount(market_info.symbol, self.order_size, pricing_proposal.sell_order_prices[0])
+            required_quote_asset_balance = pricing_proposal.buy_order_prices[0] * bid_order_size
 
         else:
             bid_order_size = market.c_quantize_order_amount(market_info.symbol, self.order_size)
             ask_order_size = market.c_quantize_order_amount(market_info.symbol, self.order_size)
+            required_quote_asset_balance = pricing_proposal.buy_order_prices[0] * (1+float(buy_fees.percent)) * bid_order_size
```

Comment:
I don't think this line is necessary. Other exchanges do not require order sizes to be adjusted for buy fees.

Binance was the only exchange that took fees out of the base asset when you buy it, so that's why we originally adjusted order sizes. We aren't taking that into account anymore, and that's what this PR does. But we don't need to adjust order sizes in other markets for fees, since they never did that in the first place.

Reply:
The PR is to address the edge case of having enough balance but not enough balance for fees, would case insufficient balance issues. For example in Coinbase, you might have enough USD for 2 ETH but since fees is taken out in USD, your order might not go through as exchange would only allow to buy 1.999 ETH. The balance calculation earlier did not include fees and now I have incorporated that. This does not affect order sizing.

Reply:
Got it - you're right that this is necessary for non-Binance exchanges.